### PR TITLE
Replace vera with clang-format

### PR DESCRIPTION
**Pull request by rerobika**

### CODE REVIEW

**Comment by rerobika** [comment]
Note: depends on #4515.

Moreover, if the style change is accepted by the maintainers, I'll update the coding standards documentation.

**Review by akosthekiss — COMMENTED** [review]
Some, feedback. I surely haven't exhaustively reviewed all changed files.

**Comment by lygstate** [comment]
I like this idea, maybe we need python script to format files automatically

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  AfterUnion:      true
+  AfterExternBlock: false
+  BeforeElse:      true
+  IndentBraces:    false
+  SplitEmptyFunction: true
+BreakBeforeBinaryOperators: NonAssignment
+BreakBeforeBraces: Custom
+BreakBeforeTernaryOperators: true
+BreakStringLiterals: true
+ColumnLimit:     120
```
[the remark this answers is not in the document]
Comment (rerobika) [reply .clang-format:36]: It is the auto generated gnu-stlye descriptor with jerry-style modifications. I didn't take effort on unify the style in the descriptor but once the approach is accepted, I'll definitely update it as well.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+CommentPragmas:  '^ IWYU pragma:'
```
[the remark this answers is not in the document]
Comment (rerobika) [reply .clang-format:37]: Nope, seems leftover.

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+PenaltyBreakBeforeFirstCallParameter: 19
+PenaltyBreakComment: 300
+PenaltyBreakFirstLessLess: 120
+PenaltyBreakString: 1000
+PenaltyBreakTemplateDeclaration: 10
+PenaltyExcessCharacter: 1000000
+PenaltyReturnTypeOnItsOwnLine: 60
+PointerAlignment: Right
+ReflowComments:  true
+SortIncludes:    true
```
[the remark this answers is not in the document]
Comment (rerobika) [reply .clang-format:63]: Sure, good idea.

File: .github/workflows/gh-actions.yml [hunk 4]
```diff
@@ -13,10 +13,10 @@ jobs:
         with:
           fetch-depth: 0
       - run: sudo apt update
-      - run: sudo apt install doxygen vera++ cppcheck pylint python-serial
+      - run: sudo apt install doxygen clang-format-10 cppcheck pylint python-serial
```
[the remark this answers is not in the document]
Comment (rerobika) [reply .github/workflows/gh-actions.yml:16]: I cannot answer for this question right now, the only I can state that the configuration was tuned according to the clang-format v10 documentation. I'll check the other versions as well.

File: jerry-core/api/jerry-snapshot.c [hunk 5]
```diff
@@ -414,19 +410,16 @@ static_snapshot_add_compiled_code (const ecma_compiled_code_t *compiled_code_p,
...
       literal_start_p[i] = 0;
     }
     else
     {
-      uint32_t offset = static_snapshot_add_compiled_code (bytecode_p,
-                                                           snapshot_buffer_p,
-                                                           snapshot_buffer_size,
-                                                           globals_p);
+      uint32_t offset =
+        static_snapshot_add_compiled_code (bytecode_p, snapshot_buffer_p, snapshot_buffer_size, globals_p);
```
Comment (zherczeg) [inline jerry-core/api/jerry-snapshot.c:422]: This looks incorrect syntax.

File: jerry-core/api/jerry-snapshot.c [hunk 6]
```diff
@@ -535,8 +527,7 @@ jerry_snapshot_set_offsets (uint32_t *buffer_p, /**< buffer */
     JERRY_ASSERT ((code_size % sizeof (uint32_t)) == 0);
     buffer_p += code_size / sizeof (uint32_t);
     size -= code_size;
-  }
-  while (size > 0);
+  } while (size > 0);
```
Comment (zherczeg) [inline jerry-core/api/jerry-snapshot.c:530]: This looks incorrect syntax.

File: jerry-core/api/jerry-snapshot.c [hunk 7]
```diff
@@ -439,8 +432,7 @@ static_snapshot_add_compiled_code (const ecma_compiled_code_t *compiled_code_p,
 
   while (literal_start_p < (ecma_value_t *) buffer_p)
   {
-    if (!ecma_is_value_direct_string (*literal_start_p)
-        && !ecma_is_value_empty (*literal_start_p))
+    if (!ecma_is_value_direct_string (*literal_start_p) && !ecma_is_value_empty (*literal_start_p))
```
Comment (zherczeg) [inline jerry-core/api/jerry-snapshot.c:435]: Why this is moved to one line? These are often harder to read.

File: jerry-core/api/jerry-snapshot.c [hunk 8]
```diff
@@ -928,8 +905,7 @@ jerry_exec_snapshot (const uint32_t *snapshot_p, /**< snapshot */
 
   const jerry_snapshot_header_t *header_p = (const jerry_snapshot_header_t *) snapshot_data_p;
 
-  if (header_p->magic != JERRY_SNAPSHOT_MAGIC
-      || header_p->version != JERRY_SNAPSHOT_VERSION
+  if (header_p->magic != JERRY_SNAPSHOT_MAGIC || header_p->version != JERRY_SNAPSHOT_VERSION
       || !snapshot_check_global_flags (header_p->global_flags))
```
Comment (zherczeg) [inline jerry-core/api/jerry-snapshot.c:909]: Incorrect syntax.

File: jerry-core/ecma/base/ecma-alloc.c [hunk 9]
```diff
@@ -99,7 +101,7 @@ ecma_dealloc_object (ecma_object_t *object_p) /**< object to be freed */
  *
  * @return pointer to allocated memory
  */
-extern inline ecma_extended_object_t * JERRY_ATTR_ALWAYS_INLINE
+extern inline ecma_extended_object_t *JERRY_ATTR_ALWAYS_INLINE
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-alloc.c:104]: Incorrect syntax.

File: jerry-core/ecma/base/ecma-errors.c [hunk 10]
```diff
@@ -20,114 +20,114 @@
 /**
  * Error message, if an argument is has an error flag
  */
-const char * const ecma_error_value_msg_p = "Argument cannot be marked as error";
+const char* const ecma_error_value_msg_p = "Argument cannot be marked as error";
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-errors.c:23]: Incorrect syntax.

File: jerry-core/ecma/base/ecma-extended-info.c [hunk 11]
```diff
@@ -13,10 +13,12 @@
  * limitations under the License.
  */
 
-#include "byte-code.h"
-#include "ecma-helpers.h"
 #include "ecma-extended-info.h"
 
+#include "ecma-helpers.h"
+
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-extended-info.c:19]: These newlines are unnecessary.

File: jerry-core/ecma/base/ecma-helpers-conversion.c [hunk 12]
```diff
@@ -129,84 +128,82 @@ ecma_round_high_to_uint64 (ecma_uint128_t *num_p)
...
-  q32 += h2 >> 32u; \
-  q32 += h3; \
-  \
-  name.lo = (q1 << 32u) | ((uint32_t) q0); \
-  name.hi = q32; \
-}
-
-#if defined (__GNUC__) || defined (__clang__)
+#define ECMA_UINT128_DIV10(name)                                                        \
+  {                                                                                     \
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-helpers-conversion.c:132]: Incorrect syntax, the slash should be kept in its place.

File: jerry-core/ecma/base/ecma-extended-info.c (hunk 11 above)
Comment (rerobika) [reply jerry-core/ecma/base/ecma-extended-info.c:19]: That's valid.
Include order separated to incluce blocks:

1. own header
2. system header
3. jerry* header
4. ecma* header
5. everything else.

File: jerry-core/ecma/base/ecma-helpers-conversion.c (hunk 12 above)
Comment (rerobika) [reply jerry-core/ecma/base/ecma-helpers-conversion.c:132]: Slash is aligned no the longest line of the macro, so still correct.

File: .github/workflows/gh-actions.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (rerobika) [reply .github/workflows/gh-actions.yml:19]: Now we expect version `>= 10.0`.

**Comment by rerobika** [comment]
The patch is ready, the 08.CODING-STARDARDS.MD needs to be updated in a follow up patch.

**Review by dbatyai — APPROVED** [review]
LGTM

**Review by robertsipka — APPROVED** [review]
LGTM

**Pull request merged**